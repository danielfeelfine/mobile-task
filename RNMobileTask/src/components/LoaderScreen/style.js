import { StyleSheet } from 'react-native'
import { colors } from '@styles/base'

const style = StyleSheet.create({
  container: {
    flex: 1,
    paddingVertical: 26,
    paddingHorizontal: 16,
    alignItems: 'center',
    backgroundColor: colors.Light,
  },
  message: {
    marginVertical: 8,
    textAlign: 'center',
    fontWeight: '300',
    fontSize: 12,
    lineHeight: 22,
    color: colors.Gray,
  },
})

export default style
