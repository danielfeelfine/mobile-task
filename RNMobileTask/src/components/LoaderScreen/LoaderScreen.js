// @flow
import React from 'react'
import { Text, View, ActivityIndicator } from 'react-native'
import PropTypes from 'prop-types'
import style from './style'

type Props = {
  message: string
}

const LoaderScreen = (props: Props) => {
  return (
    <View style={style.container}>
      <ActivityIndicator size="small" />
      <Text style={style.message}>{props.message}</Text>
    </View>
  )
}

LoaderScreen.propTypes = {
  message: PropTypes.string.isRequired
}

export default LoaderScreen
