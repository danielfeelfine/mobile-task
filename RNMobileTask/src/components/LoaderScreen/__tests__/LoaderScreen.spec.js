import React from 'react'
import { shallow } from 'enzyme'
import LoaderScreen from './../LoaderScreen'


describe('LoaderScreen component', () => {

  it('renders correctly', () => {
    const props = {
      message: 'The news sources are loading'
    }

    const wrapper = shallow(
      <LoaderScreen message={props.message} />
    )
  
    expect(wrapper).toMatchSnapshot()
  })

})
