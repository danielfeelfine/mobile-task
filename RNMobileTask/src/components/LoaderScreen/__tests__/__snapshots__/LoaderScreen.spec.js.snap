// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderScreen component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoaderScreen
    message="The news sources are loading"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="small"
        />,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#999999",
              "fontSize": 12,
              "fontWeight": "300",
              "lineHeight": 22,
              "marginVertical": 8,
              "textAlign": "center",
            }
          }
        >
          The news sources are loading
        </Text>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 26,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animating": true,
          "color": "#999999",
          "hidesWhenStopped": true,
          "size": "small",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "The news sources are loading",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#999999",
            "fontSize": 12,
            "fontWeight": "300",
            "lineHeight": 22,
            "marginVertical": 8,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "The news sources are loading",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ActivityIndicator
            animating={true}
            color="#999999"
            hidesWhenStopped={true}
            size="small"
          />,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#999999",
                "fontSize": 12,
                "fontWeight": "300",
                "lineHeight": 22,
                "marginVertical": 8,
                "textAlign": "center",
              }
            }
          >
            The news sources are loading
          </Text>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 26,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animating": true,
            "color": "#999999",
            "hidesWhenStopped": true,
            "size": "small",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "The news sources are loading",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#999999",
              "fontSize": 12,
              "fontWeight": "300",
              "lineHeight": 22,
              "marginVertical": 8,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "The news sources are loading",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
